gateway/pkg/tracing/otel.go

```diff
@@ -0,0 +1,155 @@
+package tracing
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	jaegerprop "go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+type Exporter string
+
+const (
+	JaegerExporter   Exporter = "jaeger"
+	LogExporter      Exporter = "log"
+	OTELExporter     Exporter = "otel"
+	DisabledExporter Exporter = "disabled"
+)
+
+type Shutdown func(context.Context)
+
+// Provider returns an OpenTelemetry TracerProvider configured to use
+// the Jaeger exporter that will send spans to the provided url. The returned
+// TracerProvider will also use a Resource configured with all the information
+// about the application.
+func Provider(ctx context.Context, name, version, commit string) (shutdown Shutdown, err error) {
+	exporter := Exporter(os.Getenv("OTEL_EXPORTER"))
```

Comment:
This is not a standard env variable, we could choose a new name for it if you want.